### ISSUE TITLE
tiffsave fails with tile and jpeg compression and quality ≥ 90 on MacOS (v8.15.1)

### ISSUE THREAD

**Issue by cavenel** [libvips/libvips#3855]
On vips version 8.15.1 on MacOS, the `tiffsave` function fails when used with the following options:
`vips tiffsave input.tif output.tif --tile --compression jpeg --Q 90`
It works for any compression between 0 and 89. It fails for any compression between 90 and 100.
It works without the `--tile` or with `--compression none`.
It works on other platforms than MacOS.

**To Reproduce**
1. Use any image as input (jpg, tif...)
2. Run the command `vips tiffsave input.tif output.tif --tile --compression jpeg --Q 90`
3. Get the following error:
```
VipsJpeg: Unsupported color conversion request
VipsJpeg: Unsupported color conversion request
VipsJpeg: Unsupported color conversion request
wbuffer_write: write failed
unix error: Unknown error: -1
```

**Environment**
(please complete the following information)
 - OS: both `MacOS 13.3.1 M1 Pro` and `MacOS 12.6.8 Intel`
 - Vips: 8.15.1 (installed with brew)

**Comment by jcupitt** [comment]
Hi @cavenel,

It's working for me:

```
16:13 $ vips tiffsave k2.jpg x.tif --tile --compression jpeg --Q 90
✔ ~/pics 
16:14 $ which vips
/opt/homebrew/bin/vips
✔ ~/pics 
16:14 $ vips --version
vips-8.15.1
✔ ~/pics 
16:14 $ 
```

That's an M2 mac mini with homebrew libvips. 

Could you perhaps have a non-RGB image in `input.tif`? You can only save RGB and mono images with JPEG compression.

**Comment by cavenel** [comment]
Hi @jcupitt,

That is strange, I tried with multiple input files, including jpeg input files (so RGB).
And the fact that it works with a different compression value (like `--Q 85`) is also confusing.
I will investigate more with other images tomorrow, and share one that fails.

**Comment by cavenel** [comment]
So, after some more testing, it seems like it only fails on grayscale images.
Take any image, open in GIMP, change mode to Grayscale and export as PNG or JPG. Then vips will fail to convert it to tiff with tile and compression bigger or equal to 90.
Here is an example of image that fails:
![k2_g](https://github.com/libvips/libvips/assets/6164758/1a2ead63-2d3f-446c-8dba-bb4e99dc249d)
With the command:
```
vips tiffsave k2_g.jpg x.tif --tile --compression jpeg --Q 90
```

**Comment by jcupitt** [comment]
Ah well done! Yes I see a failure too, though only on macOS:

```
12:21 $ vips tiffsave k2_mono.jpg x.tif --tile --compression jpeg --Q 90
VipsJpeg: Unsupported color conversion request
VipsJpeg: Unsupported color conversion request
VipsJpeg: Unsupported color conversion request
wbuffer_write: write failed
unix error: Unknown error: -1
✘-1 ~/pics 
```

At a guess, Q>=90 is disabling chroma subsampling, and the very recent libtiff on macOS doesn't like this for greyscale images. I'll investigate.

**Comment by jcupitt** [comment]
After some digging, it seems it's mozjpeg that's causing the problem. Here's a dockerfile that fails in the same way on ubuntu 23.20:

https://github.com/jcupitt/docker-builds/tree/master/libvips-mozjpeg-ubuntu23.10

I'll poke a bit more.

**Comment by jcupitt** [comment]
Yes, mozjpeg seems to be missing some of the colourspace conversions that the other libjpeg implementations support. I've changed the way libvips disables chroma subsample for tiff jpeg write and it seems to work now.

This fix will be in 8.15.2, thanks for the report!
